foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/actions/CountAction.java

```diff
@@ -105,4 +104,15 @@ public ActionResponse execute(CountRequest parameter) throws QueryStoreException
                     "Error running count query.", e);
         }
     }
+
+    @Override
+    protected boolean validate() {
+        getParameter().setTable(ElasticsearchUtils.getValidTableName(getParameter().getTable()));
+        try {
+            return !(getParameter().getTable() == null || !getTableMetadataManager().exists(ElasticsearchUtils.getValidTableName(getParameter().getTable())));
```

Comment:
When tableMetadataManager is throwing exception, it means something is wrong with system and hence we should throw QUERY_EXECUTION_ERROR error code. In case table is not present, throw NO_SUCH_TABLE error code.